app/controllers/v0/facilities/ccp_controller.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class V0::Facilities::CcpController < FacilitiesController
+  def show
+    ppms = Facilities::PPMSClient.new
+    # id passed in like ccp_XXXXXXXX. API call requires XXXXXXX
+    id = params[:id][4..-1]
+    result = ppms.provider_info(id)
+    render json: format_details(result)
+  end
+
+  def specialties
+    ppms = Facilities::PPMSClient.new
+    result = ppms.specialties
+    render json: result
+  end
+
+  private
+
+  def format_details(prov_info)
```

Comment:
i agree, please consider introducing a layer of abstraction here, model + serializer (view), with unit level specs. I also agree with the idea of this possibly being a separate controller, perhaps nested within if that is more sensible.

See how prescriptions/tracking was implemented for example.

Reply:
I have converted this to using ActiveModel + ActiveSerializer. This does have the unintended consequence of unavoidably wrapping the result of the endpoint in an additional layer. {data: /* object */} which means that there will have to be a tweak on the front end before this will function